packages/moonstone/VirtualList/VirtualListBase.js

```diff
@@ -441,20 +520,12 @@ const VirtualListBaseFactory = (type) => {
 
 		jumpToSpottableItem = (keyCode, target) => {
 			const
-				{cbScrollTo, dataSize, isItemDisabled} = this.props,
+				{cbScrollTo, dataSize, isItemDisabled, wrap, wrapAnimated} = this.props,
+				{findSpottableExtent, findSpottableItem, getExtentIndex} = this,
 				{firstIndex, numOfItems} = this.uiRef.state,
-				{isPrimaryDirectionVertical} = this.uiRef,
+				{dimensionToExtent, isPrimaryDirectionVertical} = this.uiRef,
 				rtl = this.props.rtl,
-				currentIndex = Number.parseInt(target.getAttribute(dataIndexAttribute));
-
-			if (
-				!isItemDisabled || // It is the case that there is no disabled items in a list.
-				isItemDisabled(currentIndex) // If the currnet index item is disabled, it means that all items in a list are disabled.
-			) {
-				return false;
-			}
-
-			const
+				currentIndex = Number.parseInt(target.getAttribute(dataIndexAttribute)),
 				isForward = (
```

Comment:
Please move the code defining `isForward` and `isBackward` after cheking `if (isItemDisabled(currentIndex)` because in the most of the cases, disabled items are used and it could make VirtualList better navigating performance with all enabled items.

Reply:
Calculations are not so hard and I think it's okay regarding key event rate and calculation time.